tasks/html2jsonld.js

```diff
@@ -39,9 +39,19 @@ module.exports = (grunt) => {
     }
   }
   
-  function setPath(url, file, anchor, selector)
+  function setPath(url, rootPath, file, anchor, selector)
   {
-    return url + path.dirname(file).split("/").slice(-1)[0] + "/" + 
+    rootPath = rootPath.replace(".\\",'').replace(/\\/g,"/") + "/";
+    var segments = path.dirname(file).replace(rootPath, '').split("/");
```

Comment:
Dunno if there are any tests for the .js but if so i'd like to see what happens if the parameters are null, undefined or empty as there doesn't seem to be any validation

Reply:
Another good shout, have added validation. No test I'm afraid. ;-(